src/builder.py

```diff
@@ -276,8 +279,26 @@ def get_new_executable(self):
         return path
 
 
-class HostBuilder:
-    def __init__(self, build_directory, install_directory):
+class Abstract(NotImplementedError):
+    pass  # TODO: use abc?
+
+
+class Builder:
+    def __init__(self, build_directory: BuildDirectory,
+                 install_directory: Optional[str]) -> None:
```

Comment:
The constructor arguments serve no purpose, right? Maybe let this be a memberless class?

Reply:
They serve to define the types of `self._build_directory` and `self._install_directory` which are referred to elsewhere.

I made `Builder` inherit from `ABC` and marked the other methods `@abstractmethod` so it is a memberless class now.